Sources/SwiftDocC/SourceRepository/SourceRepository.swift

```diff
@@ -9,33 +9,40 @@
 */
 
 import Foundation
+import ArgumentParser
 
 /// A remote repository that hosts source code.
 public struct SourceRepository {
     /// The path at which the repository is cloned locally.
     public var checkoutPath: String
-    
+
     /// The base URL where the service hosts the repository's contents.
     public var sourceServiceBaseURL: URL
-    
+
     /// A function that formats a line number to be included in a URL.
     public var formatLineNumber: (Int) -> String
-    
+
     /// Creates a source code repository.
     /// - Parameters:
     ///   - checkoutPath: The path at which the repository is checked out locally and from which its symbol graphs were generated.
     ///   - sourceServiceBaseURL: The base URL where the service hosts the repository's contents.
     ///   - formatLineNumber: A function that formats a line number to be included in a URL.
-    public init(
+    public init (
         checkoutPath: String,
         sourceServiceBaseURL: URL,
         formatLineNumber: @escaping (Int) -> String
     ) {
-        self.checkoutPath = checkoutPath
+
+        // Get the absolute path of a file without the file:// prefix because this function used to only
+        // expect absolute paths from a user and didn't convert checkoutPath to a URL and back.
+        let absoluteCheckoutPath = URL(fileURLWithPath: checkoutPath).absoluteString
+        let startIndex = absoluteCheckoutPath.index(absoluteCheckoutPath.startIndex, offsetBy: 7)
+        self.checkoutPath = String(absoluteCheckoutPath[startIndex...])
```

Comment:
I was hesitant to use `URL(fileURLWithPath: checkoutPath).path` because Xcode said it's going to be deprecated in the future. What should I do in the future when something is going to be deprecated?

Also, I've updated the code to use `.path` instead of the pattern I was using. Thank you for the advice!